rewrite-maven/src/main/java/org/openrewrite/maven/internal/MavenPomDownloader.java

```diff
@@ -516,6 +516,14 @@ public Pom download(GroupArtifactVersion gav,
                             RawPom rawPom = RawPom.parse(fis, Objects.equals(versionMaybeDatedSnapshot, gav.getVersion()) ? null : versionMaybeDatedSnapshot);
                             Pom pom = rawPom.toPom(inputPath, repo).withGav(resolvedGav);
 
+                            if("jar".equals(pom.getPackaging())){
+                                Path jar = f.toPath().getParent().resolve(gav.getArtifactId() + '-' + versionMaybeDatedSnapshot + ".jar");
+                                if (!Files.exists(jar)) {
+                                    // This is a corrupt dependency.
```

Comment:
As we've seen, this typically isn't the case. It just means that the jar has not been downloaded.